game/src/edit/new_traffic_signals.rs

```diff
@@ -0,0 +1,766 @@
+                    },
+                ])
+                .centered_vert()
+                .align_right(),
+            ]),
+        ])
+        .padding(10);
+
+        if idx == selected {
+            col.push(phase_col.bg(Color::hex("#2A2A2A")));
+        } else {
+            col.push(phase_col);
+        }
+    }
+
+    // TODO Widget::separator(ctx, pct_width)
+    col.push(
+        Widget::draw_batch(
+            ctx,
+            GeomBatch::from(vec![(
+                Color::WHITE,
+                // TODO draw_batch will scale up, but that's inappropriate here, since we're
+                // depending on window width, which already factors in scale
+                Polygon::rectangle(0.2 * ctx.canvas.window_width / ctx.get_scale_factor(), 2.0),
+            )]),
+        )
+        .centered_horiz(),
+    );
+    col.push(Btn::text_fg("Add new phase").build_def(ctx, None));
+
+    Composite::new(Widget::col(col))
+        .aligned(HorizontalAlignment::Left, VerticalAlignment::Top)
+        .exact_size_percent(30, 85)
+        .build(ctx)
+}
+
+fn draw_multiple_signals(
+    ctx: &mut EventCtx,
+    app: &App,
+    members: &BTreeSet<IntersectionID>,
+    idx: usize,
+) -> GeomBatch {
+    let mut batch = GeomBatch::new();
+    for i in members {
+        batch.push(
+            app.cs.normal_intersection,
+            app.primary.map.get_i(*i).polygon.clone(),
+        );
+
+        draw_signal_phase(
+            ctx.prerender,
+            &app.primary.map.get_traffic_signal(*i).phases[idx],
+            *i,
+            None,
+            &mut batch,
+            app,
+            TrafficSignalStyle::Sidewalks,
+        );
+    }
+
+    // Transform to a screen-space icon. How much should we scale things down?
+    batch = batch.autocrop();
+    let mut zoom: f64 = 1.0;
+    if true {
+        // Make the whole thing fit a fixed width
+        let mut bounds = Bounds::new();
+        for i in members {
+            bounds.union(app.primary.map.get_i(*i).polygon.get_bounds());
+        }
+        zoom = 300.0 / bounds.width();
+    } else {
+        // Don't let any intersection get too small
+        for i in members {
+            zoom = zoom.max(150.0 / app.primary.map.get_i(*i).polygon.get_bounds().width());
+        }
+    }
+    batch.scale(zoom)
+}
+
+impl BundleEdits {
+    fn apply(&self, app: &mut App) {
+        for s in &self.signals {
+            app.primary.map.incremental_edit_traffic_signal(s.clone());
+        }
+    }
+
+    fn get_current(app: &App, members: &BTreeSet<IntersectionID>) -> BundleEdits {
+        let signals = members
+            .iter()
+            .map(|i| app.primary.map.get_traffic_signal(*i).clone())
+            .collect();
+        BundleEdits { signals }
+    }
+
+    // If the intersections haven't been edited together before, the number of phases and the
+    // durations might not match up. Just initially force them to align somehow.
```

Comment:
Maybe related to this comment, I was surprised that this synchronization mutates the grouped signals without an explicit "save" step. 

Tried to capture in this video what I mean:

![bulk_edit mov](https://user-images.githubusercontent.com/217057/90072707-c4a73700-dcac-11ea-8c80-8b31f05f21c7.gif)

It might be hard to make sense of the gif... basically everything up until viewing and exiting the uber turn viewer is just setup and is working as expected. However, I was surprised after exiting the uber turn viewer that the other intersection had been changed even though I never clicked "save" or "finish" or whatever.

Reply:
Yep, this new editor is unfinished. I'll add in the "actually commit the changes" bit -- `apply_map_edits` -- next. In the meantime, breaking things in strange ways is a fine thing to check in, because this editor is safely hidden in dev mode. :P

Reply:
Sounds good.